src/main/java/org/apache/tomcat/jni/SSL.java

```diff
@@ -234,6 +234,9 @@
     public static final int SSL_SELECTOR_FAILURE_NO_ADVERTISE = 0;
     public static final int SSL_SELECTOR_FAILURE_CHOOSE_MY_LAST_PROTOCOL = 1;
 
+    public static final int SSL_ST_CONNECT = 0x1000;
```

Comment:
@normanmaurer - that is a good point. I'm not sure if this will be a deal breaker though...lets discuss in new issue :)